image/image.go

```diff
@@ -0,0 +1,386 @@
+		}
+		// local install
+		if info.Revision.Unset() {
+			info.Revision = snap.R(-1)
+		}
+
+		switch info.Type {
+		case snap.TypeOS:
+			bootvar = "snap_core"
+		case snap.TypeKernel:
+			bootvar = "snap_kernel"
+			if err := extractKernelAssets(fullname, info); err != nil {
+				return err
+			}
+		}
+
+		name := filepath.Base(fullname)
+		if bootvar != "" {
+			if err := bootloader.SetBootVar(bootvar, name); err != nil {
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
+func runCommand(cmdStr ...string) error {
+	cmd := exec.Command(cmdStr[0], cmdStr[1:]...)
+	if output, err := cmd.CombinedOutput(); err != nil {
+		return fmt.Errorf("cannot run %v: %s (%s)", cmdStr, err, output)
+	}
+	return nil
+}
+
+func extractKernelAssets(snapPath string, info *snap.Info) error {
+	// FIXME: hrm, hrm, we need to be root for this - alternatively
+	//        we could make boot.ExtractKernelAssets() work on
+	//        a plain .snap file again (i.e. revert bee59a2)
+	if err := os.MkdirAll(info.MountDir(), 0755); err != nil {
+		return err
+	}
+	defer os.Remove(filepath.Dir(info.MountDir()))
+	defer os.Remove(info.MountDir())
+
+	if err := runCommand("mount", snapPath, info.MountDir()); err != nil {
+		return err
+	}
+	defer runCommand("umount", info.MountDir())
+
+	pb := progress.NewTextProgress()
+	if err := boot.ExtractKernelAssets(info, pb); err != nil {
+		return err
+	}
+	return nil
+}
+
+func copyLocalSnapFile(snapName, targetDir string) (string, *snap.Info, error) {
+	snapFile, err := snap.Open(snapName)
+	if err != nil {
+		return "", nil, err
+	}
+	info, err := snap.ReadInfoFromSnapFile(snapFile, nil)
+	if err != nil {
+		return "", nil, err
+	}
+	// local snap gets sideloaded revision
+	if info.Revision.Unset() {
+		info.Revision = snap.R(-1)
+	}
+	dst := filepath.Join(targetDir, filepath.Dir(info.MountFile()))
+
+	return dst, info, osutil.CopyFile(snapName, dst, 0)
+}
+
+type downloadOptions struct {
+	Series       string
+	TargetDir    string
+	StoreID      string
+	Channel      string
+	Architecture string
+}
+
+// replaced in the tests
+var storeNew = func(storeID string) Store {
+	return store.New(nil, storeID, nil)
+}
+
+type Store interface {
+	Snap(name, channel string, devmode bool, user *auth.UserState) (*snap.Info, error)
+	Download(name string, downloadInfo *snap.DownloadInfo, pbar progress.Meter, user *auth.UserState) (path string, err error)
+}
+
+func downloadSnapWithSideInfo(name string, opts *downloadOptions) (string, *snap.Info, error) {
+	if opts == nil {
+		opts = &downloadOptions{}
+	}
+
+	if opts.Series != "" {
+		oldSeries := release.Series
+		defer func() { release.Series = oldSeries }()
+
+		release.Series = opts.Series
+	}
+	if opts.Architecture != "" {
+		oldArchitecture := arch.UbuntuArchitecture()
+		defer func() { arch.SetArchitecture(arch.ArchitectureType(oldArchitecture)) }()
+
+		arch.SetArchitecture(arch.ArchitectureType(opts.Architecture))
+	}
+
+	// FIXME: this really should be done on the server side
```

Comment:
I'd just drop this comment. I agree it'd be nice to fix, but the workaround is trivial/obvious and much easier to read than the comment itself.